sdk/keyvault/azure-security-keyvault/azure/security/keyvault/keys/_client.py

```diff
@@ -3,266 +3,715 @@
-        config.cert = config.timeout = None
-        config.verify = True
-        return config
-
-    def __init__(self, vault_url, credentials, config=None, transport=None):
-        # type: (str, Any, Configuration, HttpTransport) -> None
-        self.vault_url = vault_url.strip("/")
-        config = config or KeyClient.create_config()
-        transport = RequestsTransport(config)
+        pass  # TODO
+
+    def __init__(self, vault_url, credentials, config=None, api_version=None, **kwargs):
+        # type: (str, Any, Configuration, Optional[str], Mapping[str, Any]) -> None
+        # TODO: update type hint for credentials
+        if not credentials:
+            raise ValueError("credentials")
+
+        if not vault_url:
+            raise ValueError("vault_url")
+
+        self._vault_url = vault_url
+
+        if api_version is None:
+            api_version = KeyVaultClient.DEFAULT_API_VERSION
+
+        config = config or KeyVaultClient.get_configuration_class(api_version)(credentials)
+
+        # TODO generated default pipeline should be fine when token policy isn't necessary
         policies = [
-            config.user_agent_policy,
             config.headers_policy,
-            BearerTokenCredentialPolicy(credentials),
+            config.user_agent_policy,
+            config.proxy_policy,
+            _BearerTokenCredentialPolicy(credentials),
             config.redirect_policy,
             config.retry_policy,
             config.logging_policy,
         ]
-        self._pipeline = Pipeline(transport, policies=policies)
-        models = {
-            "DeletedKey": DeletedKey,
-            "DeletedKeyItem": DeletedKeyItem,
-            "DeletedKeyItemPaged": DeletedKeyItemPaged,
-            "JsonWebKey": JsonWebKey,
-            "Key": Key,
-            "KeyAttributes": KeyAttributes,
-            "KeyCreateParameters": KeyCreateParameters,
-            "KeyItem": KeyItem,
-            "KeyItemPaged": KeyItemPaged,
-        }
-        self._deserialize = Deserializer(models)
-        self._serialize = Serializer(models)
-
-    def create_key(self, name, key_type, size=None, key_ops=None, attributes=None, tags=None, curve=None, **kwargs):
-        # type: (str, str, Optional[int], Optional[List[str]], Any, Any, Any, Any) -> Key
-        url = "/".join([self.vault_url, "keys", name, "create"])
-        headers = {"Content-Type": "application/json; charset=utf-8", "x-ms-client-request-id": str(uuid.uuid1())}
-        create_params = KeyCreateParameters(
-            kty=key_type, key_size=size, key_ops=key_ops, key_attributes=attributes, tags=tags, curve=curve, **kwargs
+        transport = RequestsTransport(config)
+        pipeline = Pipeline(transport, policies=policies)
+
+        self._client = KeyVaultClient(credentials, api_version=self._api_version, pipeline=pipeline)
+
+    @property
+    def vault_url(self):
+        # type: () -> str
+        return self._vault_url
+
+    def _create_key(
```

Comment:
NIT: For some methods, you have every parameter in a single line, for others like [here ](https://github.com/Azure/azure-sdk-for-python/pull/5437/files#diff-1a671db84e37d28197ad85eacecd0215R108)the parameters are all in one line.
It would be good to maintain consistency across the code. I don't know if there is a standard for Python, but worth checking.

Reply:
This is something we have in python linting rules for a line length of 120. It would keep things in a single line only until reaches a line limit of 120. 
I think it would better to keep this consistency across the language as all other packages would follow the same.